### ISSUE TITLE
You could fetch in getServerSideProps instead (but you probably shouldn't? :)

### ISSUE THREAD

**Issue by joe-dev-public** [fac24/immortal-garments#43]
https://github.com/fac24/immortal-garments/blob/94a55ddf67fac0847cb56c55a58d80246b9b4f44/pages/clothes-condition/donate.js#L10

I don't think the way you're doing it above will necessarily cause any problems? But it might be better in some situations to put the fetch in getServerSideProps (to help Next do whatever optimisation it can). And then pass the results of that as a prop to the ``Donate`` function.

As [this page](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props#when-should-i-use-getserversideprops) says, you can fetch data on the client side (which I think is what you're currently doing) if you don't need to pre-render. And given your fetch is the result of user input (the search) from a form on the same page, I'm pretty sure you can't pre-render?

(The same thing applies to ``recycle.js`` as well.)

Long story short, no changes required or suggested, but just something to keep in mind 😛
